compiler-core/src/parse.rs

```diff
@@ -2167,14 +2167,27 @@ where
                         self.parse_const_record_finish(start, Some(name), upname, end)
                     }
                     Some((_, Token::Name { name: end_name }, end)) => {
-                        let _ = self.next_tok(); // upname
-                        Ok(Some(Constant::Var {
-                            location: SrcSpan { start, end },
-                            module: Some(name),
-                            name: end_name,
-                            constructor: None,
-                            typ: (),
-                        }))
+                        if self.peek_tok1() == Some(&Token::LeftParen) {
+                            let _ = self.next_tok(); // name
+
+                            parse_error(
+                                ParseErrorType::UnexpectedToken {
+                                    expected: vec!["A constant definition".to_string()],
+                                    hint: Some("Functions cannot be called in constant declarations. You can use let instead (and make sure it is inside a scope)".to_string())
```

Comment:
```suggestion
                                    hint: Some("Functions can only be called within other functions.".to_string())
```

Perhaps this might be clearer? I wasn't sure what the `let` and scope bits meant.

Reply:
Yes, I think that's way clearer. I was trying to make a hint related to what was described in the issue, which mentions that if the function is called as `let a =` inside a function it works. But it is better to have this be more generic.